services/director-v2/src/simcore_service_director_v2/models/schemas/constants.py

```diff
@@ -8,8 +8,11 @@
 DYNAMIC_SIDECAR_SERVICE_PREFIX = "dy-sidecar"
 DYNAMIC_PROXY_SERVICE_PREFIX = "dy-proxy"
 
-DYNAMIC_SIDECAR_DOCKER_IMAGE_RE = (
-    r"(^(local|itisfoundation)/)?(dynamic-sidecar):([\w]+)"
-)
+# This matches registries by:
+# - local
+# - itisfoundation
+# - 10.0.0.0:8473 (IP & Port)
+DYNAMIC_SIDECAR_DOCKER_IMAGE_RE = r"(^(local|itisfoundation|((([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]):[0-9]+))/)?(dynamic-sidecar):([\w]+)"
```

Comment:
I'll let PC confirm this one - he dreams in regex

Reply:
Sorry but this is way convoluted for what we need here.

Please use below expression:
```python
r"(^([a-zA-Z0-9:.-]+)/)?(dynamic-sidecar):([a-zA-Z0-9.]+)"
```

Add tests for dealing with the following cases:
- These should pass the regex
```
local/dynamic-sidecar:sadasd
itisfoundation/dynamic-sidecar:sadasd
10.10.10.10.no.ip:8080/dynamic-sidecar:10.0.1
10.10.10.10.no-ip:8080/dynamic-sidecar:sadasd
```
- this fails (note: `_` is not allowed in dns)
```
10.10.10.10.no_ip:8080/dynamic-sidecar:sadasd
10.10.10.10.no.ip:8080/dynamic-sidecar:the_tag
```

Reply:
Tests added, regex change, please re-review thx